catalog/src/main/java/it/eng/catalog/rest/api/ProxyAPIController.java

```diff
@@ -0,0 +1,54 @@
+package it.eng.catalog.rest.api;
+
+import java.util.List;
+
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.fasterxml.jackson.databind.JsonNode;
+
+import it.eng.catalog.model.Catalog;
+import it.eng.catalog.service.ProxyAPIService;
+import it.eng.tools.controller.ApiEndpoints;
+import it.eng.tools.model.Serializer;
+import it.eng.tools.response.GenericApiResponse;
+import lombok.extern.slf4j.Slf4j;
+
+@RestController
+@RequestMapping(consumes = MediaType.APPLICATION_JSON_VALUE, produces = MediaType.APPLICATION_JSON_VALUE, path = ApiEndpoints.PROXY_V1)
+@Slf4j
+public class ProxyAPIController {
+
+	private final ProxyAPIService proxyApiService;
```

Comment:
We also need the proxy endpoint to fetch the dataset that we want ( to forward requests to https://provider.com/catalog/datasets/{id} )

Reply:
"Problem" here is that okHttpClient we use, does not support GET request with body, like defined in DSP protocol.
There are some hacks how to do that, but that did not solved solution. Maybe I did not verified it correct.